src/main/java/mate/academy/springbootwebgreqit/repository/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,32 @@
+package mate.academy.springbootwebgreqit.repository;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.springbootwebgreqit.dto.BookSearchParameters;
+import mate.academy.springbootwebgreqit.model.Book;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationBuilder implements SpecificationBuilder<Book>{
+    private static final String ISBN_KEY = "isbn";
+    private static final String TITLE_KEY = "title";
+    private static final String AUTHOR_KEY = "author";
+
+    private final BookSpecificationProviderManager bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParameters) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.isbns() != null && searchParameters.isbns().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider(ISBN_KEY).getSpecification(searchParameters.isbns()));
+        }
+        if (searchParameters.titles() != null && searchParameters.titles().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider(TITLE_KEY).getSpecification(searchParameters.titles()));
+        }
+        if (searchParameters.authors() != null && searchParameters.authors().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager.getSpecificationProvider(AUTHOR_KEY).getSpecification(searchParameters.authors()));
```

Comment:
that's a very long line, it couldn't pass 100 chars/line check in checkstyle plugin